app/services/current-route.js

```diff
@@ -0,0 +1,16 @@
+import Ember from 'ember';
+
+const {
+  getOwner,
+  Service,
+  set
+} = Ember;
+
+export default Service.extend({
+  currentRouteName: null,
+
+  updateCurrentRouteName() {
+    let { currentPath } = getOwner(this).lookup('controller:application');
+    set(this, 'currentRouteName', currentPath);
+  }
```

Comment:
This works fine, but it's a bit of a confusing way to just set a property on an object, which this service really is. It's an object, with a single property, which is a singleton in the context of our application. I'm thinking, why not just have the routes that call `get(this, 'currentRoute').updateCurrentRouteName()`  instead just call something like 

```Javascript
set(this, 'currentRoute.currentRouteName', 'some.new.name');
```

While we're at it, since the name of the service is already `currentRoute`, why not call the property `currentRoute.name`, instead of the redundant `currentRoute.currentRouteName`.

Lastly, and this is a big one, there is a router service in ember, though it's still private right now:

It's deinfed [here](https://github.com/emberjs/ember.js/blob/master/packages/ember-routing/lib/services/routing.js#L26) and is used by the `LinkComponent` [here](https://github.com/emberjs/ember.js/blob/v2.12.0/packages/ember-glimmer/lib/components/link-to.js#L528).

Basically, we can replace the usage of our own currentRoute service by simply replacing injections of it with 

```Javascript
routing: service('-routing'),
```

We can name it `currentRoute` to, if we want.

We can also then alias the `currentRouteName` easily:

```Javascriot
currentRouteName: alias('router.currentRouteName')
```